core/src/main/scala/cats/Alternative.scala

```diff
@@ -99,8 +100,15 @@ import simulacrum.typeclass
       val F = self
       val G = Applicative[G]
     }
+
+  def fromIterableOnce[A](as: IterableOnce[A]): F[A] =
+    combineAllK(as.iterator.map(pure(_)))
+
+  final def fromFoldable[G[_]: Foldable, A](as: G[A]): F[A] =
```

Comment:
I may have suggested making this final, but on second thought I think it should not be.

It could be that F is very efficient to build from a foldRight or a foldLeft or a foldMap. Since we should assume that the foldable for G knows the most efficient way to do those, the `F` implementation here may want to call one of those rather than converting to an `Iterable[A]` and going from there.

This is probably minor, and maybe never a big deal, but the `final` will block this (I may have suggested or previously agreed with final, if so, apologies for the flip-flop).

Reply:
Oh yeah, that's a good point.

FTR this was my suggestion in https://github.com/typelevel/cats/pull/4084#issuecomment-1024384440.

The problem is we can't currently check this implementation in the laws; see https://github.com/typelevel/cats/pull/4084#issuecomment-1024375090.

So there's a bit of performance vs yolo factor here 😉 

> This is probably minor, and maybe never a big deal, but the `final` will block this

We can always take away a `final` modifier, but we can't add it back. So I'd be fine to leave this until a practical need to change this comes up.

Reply:
why can't we check the laws? why can't we check that `fromFoldable(g) == fromIterable(g.toIterable)` in the laws?

Reply:
Yeah AFAICT we would have to add extra type parameters to `AlternativeTests#alternative` to do this. Unless I've missed something?